src/main/java/com/sarality/sync/APISyncNewDataFetcher.java

```diff
@@ -0,0 +1,42 @@
+package com.sarality.sync;
+
+import com.sarality.db.Column;
+import com.sarality.db.Table;
+import com.sarality.db.query.Operator;
+import com.sarality.db.query.Query;
+import com.sarality.db.query.SimpleQueryBuilder;
+import com.sarality.db.query.SimpleQueryClause;
+
+import java.util.List;
+
+import hirondelle.date4j.DateTime;
+
+/**
+ * Sync data fetcher for new records - which were created after last sync time
+ *
+ * @author Satya@ (Satya Puniani)
+ */
+
+public class APISyncNewDataFetcher<T> implements APISyncDataFetcher<T> {
+
+  private final Table<T> table;
+  private final Column globalIdColumn;
+  private final Column timeStampColumn;
+
+  public APISyncNewDataFetcher(Table<T> table, Column globalIdColumn, Column timeStampColumn) {
+    this.table = table;
+    this.globalIdColumn = globalIdColumn;
+    this.timeStampColumn = timeStampColumn;
+  }
+
+
+  @Override
+  public List<T> fetch(DateTime syncDate) {
+    Query newQuery = new SimpleQueryBuilder()
+        .withFilter(new SimpleQueryClause(globalIdColumn, Operator.IS_NULL))
+        .withFilter(timeStampColumn, Operator.GREATER_THAN_EQUAL_TO, syncDate)
```

Comment:
Wondering if it make sense to take two timestamps in here to avoid race conditions. We may need to define the nextSyncTimestamp and make sure that we get the date where creationTimestamp column is between lastSyncTimestamp and nextSyncTimestamp.